api/dataset.go

```diff
@@ -498,21 +514,32 @@ func createNewVersionDoc(currentVersion *models.Version, version *models.Version
 	return version
 }
 
-func (api *DatasetAPI) updateDataset(id string, version *models.Version) error {
+func (api *DatasetAPI) publishDataset(id string, version *models.Version) error {
 	currentDataset, err := api.dataStore.Backend.GetDataset(id)
 	if err != nil {
 		log.ErrorC("unable to update dataset", err, log.Data{"dataset_id": id})
 		return err
 	}
 
+	var accessRights string
+
+	if currentDataset.Next.Links != nil {
+		if currentDataset.Next.Links.AccessRights != nil {
+			accessRights = currentDataset.Next.Links.AccessRights.HRef
+		}
+	}
```

Comment:
Could this logic to check the access rights field be moved to the `createNewVersionDoc` function, or is it specific to the publish dataset step?

Reply:
It is specific to a dataset. The link will not exist against a version.